include/EVT/dev/encoder.hpp

```diff
@@ -0,0 +1,112 @@
+#ifndef _EVT_ENCODER_
+#define _EVT_ENCODER_
+
+#include <EVT/io/GPIO.hpp>
+
+namespace EVT::core::DEV {
+class Encoder {
+public:
+    /**
+     * Constructs an Encoder instance. Call update() within the main loop to update encoder values.
+     * @param a a pin of the encoder
+     * @param b b pin of the encoder
+     * @param range range of the encoder positions (should be positive)
+     * @param initialPosition initial position the encoder is in
+     */
+    Encoder(IO::GPIO& a, IO::GPIO& b, int64_t range, int64_t initialPosition);
+
+    /**
+     * An Enum that represents the direction the encoder is being spun in.
+     * The direction should reset to Static after a couple of reads with no changes.
+     */
+    enum Direction {
+        Left = -1,
+        Static = 0,
+        Right = 1
+    };
+
+    /**
+     * Reads and updates the encoder rotation and the absolute position.
+     * @return what position was changed by, should be in this range: [-3,3]
+     */
+    int8_t update();
+
+    /**
+     * returns the current absolute position
+     * @return the current position of the encoder, between -range and range, inclusive
+     */
+    int64_t getPosition();
+
+    /**
+     * returns the noChangeCounter value;
+     * @return the noChangeCounter value
+     */
+    uint8_t getNoChangeCounter();
+
+    int8_t getRelativePosition();
+
+private:
+    ///GPIO pin a
+    IO::GPIO& a;
+    ///GPIO pin b
+    IO::GPIO& b;
+
+    ///position can be between = [-range, range]
+    int64_t range;
+    ///the current absolute position of the encoder
+    int64_t position;
+    ///the current direction of travel of the encoder
+    Direction currentDirection;
+    ;///counter for how many updates have occured since the encoder was moved
+    uint8_t noChangeCounter;
+    ///the current relative position of the encoder, in the range: [0,3]
+    int8_t currentRelPos;
+    ///how many updates with no change in the encoder value can occur before the direction resets
+    const static int8_t noChangeCap = 2;
+
+    /**
+     * helper method to convert binary pin values to the relative rotation of the encoder, in the range [0,3]
```

Comment:
I prefer capitalizing the first letter of each comment line, but this is not consistent across our code